src/box/alter.cc

```diff
@@ -4153,9 +4155,85 @@ replica_def_new_from_tuple(struct tuple *tuple, struct region *region)
 		diag_set(ClientError, ER_INVALID_UUID, tt_uuid_str(&def->uuid));
 		return NULL;
 	}
+	if (tuple_field_tt_hostname(&def->name, tuple, BOX_CLUSTER_FIELD_NAME,
+				    "_cluster.name") != 0)
```

Comment:
Thanks for noticing. I kept `ER_FIELD_TYPE`, but changed the error message. For `_cluster` I kept the old messages, they were fine. The `_schema` errors I changed. Now it looks as follows:
```
tarantool> box.space._cluster:replace{1, '37a7e893-4a82-493a-b50c-ddbba9ab8449', 'ba ba'}
---
- error: 'Tuple field _cluster.name type does not match one required by operation:
    expected a valid name, got a bad name'
...

tarantool> box.space._schema:replace{'replicaset_name', 'ba ba'}
---
- error: 'Tuple field _schema[''replicaset_name''].value type does not match one required
    by operation: expected a valid name, got a bad name'
...

tarantool> box.space._schema:replace{'cluster_name', ' '}
---
- error: 'Tuple field _schema[''cluster_name''].value type does not match one required
    by operation: expected a valid name, got a bad name'
...
```

I am trying to avoid adding new error codes because the more we have, the harder it is to use them for anything except logging. We already struggle with ER_TIMEOUT code and TimedOut error type. As an example of anti-pattern of adding new error codes one might look at Windows, which has 16k error codes reserved (have no idea how many are in use), multiple error codes basically mean the same, like just grep by "NOT_FOUND" in those codes. That is without counting WSA error codes. And compare with errno, all codes of which literally fit on a single man page. I would rather want our codes to be more compact.

Reply:
Thanks, looks better this time.

I see your point about not adding too many similar error codes. Looks convincing.